TESTS/mbed_hal/lp_us_tickers/main.cpp

```diff
@@ -13,98 +13,416 @@
  */
-void test_ticker_init(void)
+void ticker_init_test()
 {
+    intFlag = 0;
+
     intf->init();
-    uint32_t previous = intf->read();
+
+    /* Wait a while - let the ticker to count. */
+    wait_ms(1);
+
+    const uint32_t ticks_start = intf->read();
+
+    intf->set_interrupt(ticks_start + TICKER_INT_VAL);
+
+    /* Re-initialise the ticker. */
+    intf->init();
+
+    const uint32_t ticks_after_reinit = intf->read();
+
+    /* Wait long enough to fire ticker interrupt (should not be fired). */
+    while (intf->read() < (ticks_start + 2 * TICKER_INT_VAL)) {
+        /* Just wait. */
+    }
+
+    TEST_ASSERT(intf->read() >= (ticks_start + 2 * TICKER_INT_VAL));
+    TEST_ASSERT(ticks_start > ticks_after_reinit);
+    TEST_ASSERT_EQUAL(0, intFlag);
+}
+
+/* Test that ticker frequency is non-zero and counter is at least 8 bits */
+void ticker_info_test(void)
+{
+    const ticker_info_t* p_ticker_info = intf->get_info();
+
+    TEST_ASSERT(p_ticker_info->frequency != 0);
+    TEST_ASSERT(p_ticker_info->bits >= 8);
+}
+
+/* Test that ticker interrupt fires only when the ticker counter increments to the value set by ticker_set_interrupt. */
+void ticker_interrupt_test(void)
+{
+    intFlag = 0;
 
     intf->init();
-    uint32_t current = intf->read();
-    TEST_ASSERT_TRUE_MESSAGE(previous <= current, "init() changed the counter");
 
-    previous = intf->read();
+    const uint32_t tick_count = intf->read();
+
+    /* Set interrupt. Interrupt should be fired when tick count is equal to:
+     * tick_count + TICKER_INT_VAL. */
+    intf->set_interrupt(tick_count + TICKER_INT_VAL);
+
+    /* Wait until ticker count reach value: tick_count + TICKER_INT_VAL - TICKER_DELTA.
+     * Interrupt should not be fired. */
+    while (intf->read() < (tick_count + TICKER_INT_VAL - TICKER_DELTA)) {
+        /* Indicate failure if interrupt has fired earlier. */
+        TEST_ASSERT_EQUAL_INT_MESSAGE(0, intFlag, "Interrupt fired too early");
+    }
+
+    /* Wait until ticker count reach value: tick_count + TICKER_INT_VAL + TICKER_DELTA.
+     * Interrupt should be fired after this time. */
+    while (intf->read() < (tick_count + TICKER_INT_VAL + TICKER_DELTA)) {
+        /* Just wait. */
+    }
+
+    TEST_ASSERT_EQUAL(1, intFlag);
+
+    /* Wait until ticker count reach value: tick_count + 5 *TICKER_INT_VAL.
+     * Interrupt should not be triggered again. */
+    while (intf->read() < (tick_count + 5 * TICKER_INT_VAL)) {
+        /* Just wait. */
+    }
+
+    TEST_ASSERT_EQUAL(1, intFlag);
+}
+
+/* Test that ticker interrupt is not triggered when ticker_set_interrupt */
+void ticker_past_test(void)
+{
+    intFlag = 0;
+
     intf->init();
-    current = intf->read();
-    TEST_ASSERT_TRUE_MESSAGE(previous <= current, "init() changed the counter");
+
+    const uint32_t tick_count = intf->read();
+
+    /* Set interrupt tick count to value in the past.
+     * Interrupt should not be fired. */
+    intf->set_interrupt(tick_count - TICKER_DELTA);
+
```

Comment:
It might be good to delay for a few ticker clock cylces before intFlag is checked.

Reply:
fixed.